From e3ff52adec249b7cfa1a375866c268485a1cd344 Mon Sep 17 00:00:00 2001
From: Antonio Larrosa <larrosa@kde.org>
Date: Thu, 10 Nov 2016 15:33:36 +0100
Subject: Unsets XDG_RUNTIME_DIR in processes run with kdesu

Summary:
If we keep a XDG_RUNTIME_DIR value from another user then all kind of
breakages might happen. For example, if user foo runs
"kdesu -u bar dolphin", then when dolphin tries to get
QStandardPaths::writableLocation(RuntimeLocation), it returns an empty string
(because XDG_RUNTIME_DIR is not correct). which makes kio try
to write a socket to access kio slaves in / . Which, for most users
results in applications not being able to use kio (so no usable dolphin)
and for root, it means sockets were created in / .

Look at https://bugzilla.opensuse.org/show_bug.cgi?id=1009510 and
https://phabricator.kde.org/D3317 for more information

Reviewers: dfaure

Reviewed By: dfaure

Differential Revision: https://phabricator.kde.org/D3331
---
 src/kdesu_stub.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/kdesu_stub.c b/src/kdesu_stub.c
index dcc76d8..49db577 100644
--- a/src/kdesu_stub.c
+++ b/src/kdesu_stub.c
@@ -258,6 +258,8 @@ int main()
         unsetenv("LC_ALL");
     }
 
+    unsetenv("XDG_RUNTIME_DIR");
+
     /* Do we need to change uid? */
 
     pw = getpwnam(params[P_USER].value);
-- 
cgit v0.11.2

